src/NServiceBus.AmazonSQS/Configure/EventToTopicsMappings.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    class EventToTopicsMappings
+    {
+        public void Add(Type eventType, string[] topicsNames)
+        {
+            if (!eventsToTopicsMappings.TryGetValue(eventType, out var mapping))
+            {
+                mapping = new HashSet<string>();
+                eventsToTopicsMappings.Add(eventType, mapping);
+            }
+
+            foreach (var topicName in topicsNames)
+            {
+                if (!mapping.Contains(topicName))
+                {
+                    mapping.Add(topicName);
+                }
+            }
+        }
+
+        public bool HasMappingsFor(Type eventType)
+        {
+            return eventsToTopicsMappings.ContainsKey(eventType);
+        }
+
+        public string[] GetMappedTopicsNames(Type eventType)
+        {
+            return eventsToTopicsMappings[eventType].ToArray();
```

Comment:
I would just return  the HashSet or one of the interfaces implemented by the hashset instead of penalizing ourself with an array alloc everytime we call this method